File: content/snippets/go/interview_lcci_05_02_binary_number_to_string.go
package main

import (
	"fmt"
	"strings"
)

func binaryToString(n float64) string {
	var sb strings.Builder
	sb.WriteString("0.")
	for n > 0 {
		if sb.Len() > 32 {
			return "ERROR"
		}
		n *= 2
		if n >= 1 {
			sb.WriteByte('1')
			n -= 1
		} else {
			sb.WriteByte('0')
		}
	}
	return sb.String()
}

func main() {
	fmt.Println(binaryToString(0.625))
	fmt.Println(binaryToString(0.1))
}

File: content/snippets/typescript/interview_lcci_05_02_binary_number_to_string.ts
function binaryToString(n: number): string {
  let result = "0.";
  while (n > 0) {
    if (result.length > 32) return "ERROR";
    n *= 2;
    if (n >= 1) {
      result += "1";
      n -= 1;
    } else {
      result += "0";
    }
  }
  return result;
}

console.log(binaryToString(0.625));
console.log(binaryToString(0.1));
